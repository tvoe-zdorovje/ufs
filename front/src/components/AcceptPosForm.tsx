import React from 'react'
import {
  Alert,
  Form,
  Button,
  Row,
  Col
} from 'antd';
const FormItem = Form.Item;

const styles = require('./Shared.less')

export interface AcceptPosFormProps {
  confirmOperation: Function,
  amount: number | string,
  message: string,
  taskId: string,
}

export default class AcceptPosForm extends React.Component<AcceptPosFormProps, any> {
  render() {
    const { taskId, amount, confirmOperation } = this.props;
    return (
      <div>
        <Row>
          <Col span={8} />
          <Col span={8}>
            <Form>
              <FormItem>
                <Alert
                  message="Подтверждение операции клиентом на POS-терминале"
                  description={`Подтвердите операцию на сумму ${amount} руб. 00 коп.`}
                  type="info"
                  showIcon
                />
              </FormItem>
              <FormItem className={styles.rightAligned}>
                <Button
                  className={styles.redButton}
                  onClick={() => confirmOperation(taskId, false)}
                >
                  Отмена
                </Button>
                <Button
                  className={styles.greenButton}
                  onClick={() => confirmOperation(taskId, true)}
                >
                  Подтвердить
                </Button>
              </FormItem>
            </Form>
          </Col>
          <Col span={8} />
        </Row>
      </div>
    );
  }
}
