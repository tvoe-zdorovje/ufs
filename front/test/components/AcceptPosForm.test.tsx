import React from 'react'
import {
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'
import {
  Alert,
  Button
} from 'antd'

import AcceptPosForm, { AcceptPosFormProps } from 'components/AcceptPosForm'

describe('Компонент AcceptPosForm', () => {
  const message = 'Подтвердите операцию на сумму 100 руб. 00 коп.'
  const props: AcceptPosFormProps = {
    amount: '100',
    confirmOperation: (onComplete, onCancel) => {
    },
    message,
    taskId: '',
  }

  const wrapper = shallow(<AcceptPosForm {...props} />)
  const renderWrapper = render(<AcceptPosForm {...props} />)

  it('использует свойства <AcceptPosForm />', () => {
    expect(wrapper.find(Alert).prop('description')).toBe(message)
  })

  it('отображает 2 кнопки <Button />', () => {
    expect(wrapper.find(Button)).toHaveLength(2)
  })

  it('работает кнопка Подтвердить', () => {
    const spy = jest.fn()
    wrapper.setProps({ confirmOperation: spy })
    wrapper.update()
    wrapper.find(Button).at(0).simulate('click', {
      preventDefault() {
      }
    })
    expect(spy).toBeCalled()
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})



