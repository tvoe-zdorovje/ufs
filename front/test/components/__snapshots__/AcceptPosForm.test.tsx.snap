// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент AcceptPosForm совпадает со своим снимком 1`] = `
<div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col-8"
    />
    <div
      class="ant-col-8"
    >
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control "
            >
              <div
                class="ant-alert ant-alert-info ant-alert-with-description"
                data-show="true"
              >
                <i
                  class="anticon anticon-info-circle-o ant-alert-icon"
                />
                <span
                  class="ant-alert-message"
                >
                  Подтверждение операции клиентом на POS-терминале
                </span>
                <span
                  class="ant-alert-description"
                >
                  Подтвердите операцию на сумму 100 руб. 00 коп.
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item rightAligned"
        >
          <div
            class="ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control "
            >
              <button
                class="ant-btn ant-btn-lg redButton"
                type="button"
              >
                <span>
                  Отмена
                </span>
              </button>
              <button
                class="ant-btn ant-btn-lg greenButton"
                type="button"
              >
                <span>
                  Подтвердить
                </span>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="ant-col-8"
    />
  </div>
</div>
`;
